### PR TITLE
docs: add auth glossary

### PR DESCRIPTION
**Pull request by hf**
Add some glossary terms for auth.

### CODE REVIEW

**Review by J0 — APPROVED** [review]
lgtm

**Review by kangmingtay — APPROVED** [review]

File: apps/reference/docs/glossary.mdx [hunk 1]
```diff
@@ -8,18 +8,194 @@ Definitions for terminology and acronyms used in the Supabase documentation.
...
+tokens only once. Once a refresh token is exchanged it becomes invalid, and
+can't be exchanged again. In practice, though, a refresh token can be exchanged
+multiple times but in a short time window.
 
+#### Provider tokens
+
+A provider token is a long-lived token issued by a third-party identity
+provider. These are issued by social login services (Google, Twitter, Apple,
+Microsoft, ...) and uniquely identify a user on those platforms.
+
```
Comment (kangmingtay) [inline apps/reference/docs/glossary.mdx:34]: Can you also define what a "Provider refresh token" is here? 

```suggestion
#### Provider refresh token

A provider refresh token is a refresh token issued by a third-party identity provider which can be used to refresh the provider token returned. 
```

File: apps/reference/docs/glossary.mdx [hunk 2]
```diff
@@ -8,18 +8,194 @@ Definitions for terminology and acronyms used in the Supabase documentation.
...
-### Refresh token
+A refresh token is a long-lived (in most cases with an indefinite lifetime)
+token that is meant to be stored and exchanged for a new refresh and access
+tokens only once. Once a refresh token is exchanged it becomes invalid, and
+can't be exchanged again. In practice, though, a refresh token can be exchanged
+multiple times but in a short time window.
 
+#### Provider tokens
+
+A provider token is a long-lived token issued by a third-party identity
```
Comment (kangmingtay) [inline apps/reference/docs/glossary.mdx:31]: hmm i thought provider tokens are typically short-lived?

File: apps/reference/docs/glossary.mdx [hunk 3]
```diff
@@ -8,18 +8,194 @@ Definitions for terminology and acronyms used in the Supabase documentation.
...
+3. Confirming that users have some biological features. For example: a
+fingerprint, a certain facial structure, an iris print.
+
+### Authorization
+
+Authorization (often abbreviated authz.) is the process of verifying if a
+certain identity is allowed to access resources. Authorization often occurs by
+verifying an access token.
+
+### Session
```
Comment (kangmingtay) [inline apps/reference/docs/glossary.mdx:53]: should we also mention here that refreshing the access token will lead to continuation of the existing session?

File: apps/reference/docs/glossary.mdx [hunk 4]
```diff
@@ -8,18 +8,194 @@ Definitions for terminology and acronyms used in the Supabase documentation.
...
+### One-time password (OTP)
+
+A one-time password is a short, randomly generated and difficult to guess
+password or code that is sent to a device (like a phone number) or generated by
+a device or application.
+
+### Time-based one-time password (TOTP)
+
+A time-based one-time password is a one-time password generated at regular time
+intervals from a secret, usually from an application in a mobile device.
```
Comment (kangmingtay) [inline apps/reference/docs/glossary.mdx:75]:
```suggestion
intervals from a secret, usually from an application in a mobile device (e.g. Google Authenticator, 1Password, etc...).
```

File: apps/reference/docs/glossary.mdx (hunk 2 above)
Comment (hf) [reply apps/reference/docs/glossary.mdx:31]: Yeah I guess I mistyped.

**Review by kiwicopple — APPROVED** [review]

**Pull request merged**